src/main/java/org/candlepin/subscriptions/tally/TallySnapshotNaturalKey.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2021 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.tally;
+
+import org.candlepin.subscriptions.db.model.Granularity;
+import org.candlepin.subscriptions.db.model.ServiceLevel;
+import org.candlepin.subscriptions.db.model.TallySnapshot;
+import org.candlepin.subscriptions.db.model.Usage;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.time.OffsetDateTime;
+
+/**
+ * The business key for a TallySnapshot.
+ */
+@Data
+@AllArgsConstructor
```

Comment:
No lombok Builder?

Reply:
I'm still of the philosophy of only including what you're going to use.

Reply:
Right, but you could use it.  I see the constructor syntax elsewhere.

Reply:
You need an all args constructor for a builder, but you don't have to have a builder to use an all args constructor.

Reply:
@dmckoske let's revisit general patterns for lombok later on, for now since this PR doesn't need it.